crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.move

```diff
@@ -0,0 +1,289 @@
+        id: UID,
+        value: u64
+    }
+
+    struct Any {}
+
+    public entry fun mint_any<Any>(v: u64, ctx: &mut TxContext) {
+        transfer::transfer(
+            ObjAny<Any> {
+                id: object::new(ctx),
+                value: v,
+            },
+            tx_context::sender(ctx),
+        )
+    }
+
+    public entry fun mint_another_any<Any>(v: u64, ctx: &mut TxContext) {
+        transfer::transfer(
+            AnotherObjAny<Any> {
+                id: object::new(ctx),
+                value: v,
+            },
+            tx_context::sender(ctx),
+        )
+    }
+
+    public entry fun mint_child_any<Any>(v: u64, parent: &mut ObjAny<Any>, ctx: &mut TxContext) {
+        transfer::transfer_to_object(
+            ObjAny<Any> {
+                id: object::new(ctx),
+                value: v,
+            },
+            parent,
+        )
+    }
+
+    public entry fun mint_shared_any<Any>(v: u64, ctx: &mut TxContext) {
+        transfer::share_object(
+            ObjAny<Any> {
+                id: object::new(ctx),
+                value: v,
+            }
+        )
+    }
+
+    public entry fun obj_vec_destroy_any<Any>(v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        assert!(vector::length(&v) == 1, 0);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 42, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun two_obj_vec_destroy_any<Any>(v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        assert!(vector::length(&v) == 2, 0);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 42, 0);
+        object::delete(id);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 7, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun same_objects_any<Any>(o: ObjAny<Any>, v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        let ObjAny<Any> {id, value} = o;
+        assert!(value == 42, 0);
+        object::delete(id);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 42, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun same_objects_ref_any<Any>(o: &ObjAny<Any>, v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        assert!(o.value == 42, 0);
+        let ObjAny<Any> {id, value: _} = vector::pop_back(&mut v);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun child_access_any<Any>(child: ObjAny<Any>, v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        let ObjAny<Any> {id, value} = child;
+        assert!(value == 42, 0);
+        object::delete(id);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 42, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+}
+// "positive" tests start here
```

Comment:
I'd probably split these up into separate files with some comments. As it is, I find it hard to determine what exactly is being tested here.